The include path <sys/poll.h> is incorrect and packages
including socket_types.hpp fail to build with musl libc,
if the -Werror option is active, i.e. treat warnings as errors.

--- boost/asio/detail/socket_types.hpp	2015-03-23 21:24:12.000000000 +0100
+++ boost/asio/detail/socket_types.hpp	2015-07-30 13:10:47.719440415 +0200
@@ -54,7 +54,7 @@
 #else
 # include <sys/ioctl.h>
 # if !defined(__SYMBIAN32__)
-#  include <sys/poll.h>
+#  include <poll.h>
 # endif
 # include <sys/types.h>
 # include <sys/stat.h>
